analysis/quantifier-comparisons/05-gene-level-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,398 @@
+    cr_ka_cor = cor(`cellranger_pre-mRNA`, `kallisto_pre-mRNA`, method = "spearman")
+  )
+```
+Here, it looks like alevin-fry, alevin-fry-sketch, and alevin-fry-unfiltered-sketch all have fairly high values. 
+Alevin-fry-unfiltered was only run with the cDNA index which is probably why it has such a low correlation. 
+
+```{r}
+ggplot(nucleus_feature_qc_common_cor, aes(x = `cellranger_pre-mRNA`, y = `alevin_fry_unfiltered_sketch_pre-mRNA`)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  theme_classic()
+```
+There appears to be a lot more variation in the single nuclei samples than in the single cell samples, even with high correlation.
+
+## Correlation of Percentage of Cells gene is detected in 
+
+```{r}
+cell_detected_qc_common_cor <- feature_qc_common %>%
+  dplyr::filter(seq_unit == "cell") %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = tool,
+                     values_from = detected) %>%
+  # drop rows with NA values to ease correlation calculations
+  tidyr::drop_na()
+
+nucleus_detected_qc_common_cor <- feature_qc_common %>%
+  dplyr::filter(seq_unit == "nucleus") %>%
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = c("tool", "index_type"),
+                     values_from = detected) %>%
+  tidyr::drop_na()
+```
+
+
+```{r}
+cell_detected_qc_common_cor %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_al_fry_unfiltered_cor = cor(cellranger, alevin_fry_unfiltered, method = "spearman"), 
+    cr_al_fry_unfiltered_sketch_cor = cor(cellranger, alevin_fry_unfiltered_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+Again, these values are fairly high. It actually looks like kallisto may have the highest correlation here? 
+
+```{r}
+ggplot(cell_detected_qc_common_cor, aes(x = cellranger, y = kallisto)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  theme_classic()
+```
+You can see that in SCPCR000006, where kallisto had the lowest correlation, there is increased variance. 
+
+```{r}
+ggplot(cell_detected_qc_common_cor, aes(x = cellranger, y = alevin)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  theme_classic()
+```
+Alevin doesn't look as correlated as kallisto does here, at least for 126 and 127. 
+We see almost a second group that has higher coverage in cellranger and is off the main diagonal.
```

Comment:
This is interesting, and makes me want to see these plots for alevin-fry and sketch as well. Is this second group a common feature, or is it unique to alevin? I also think this is a place where implementing the `--cr-like` option in alevin-fry would be neat to look at in the future, if we go the alevin-fry route.

Reply:
It was a bit weird to see and is only unique to alevin. None of the alevin-fry combinations show it with all of them showing only one strong diagonal with no extra groups. I'm not entirely sure what the explanation is... but I do think as we narrow down tools, if we continue to use alevin-fry we should incorporate `--cr-like` as you said.